src/plugins/intel_cpu/src/graph_optimizer.cpp

```diff
@@ -3186,5 +3190,94 @@ void GraphOptimizer::MatchSdpaKvCache(Graph &graph) {
     }
 }
 
+void GraphOptimizer::DropRedundantMemoryOutput(Graph &graph) {
+    // When we have a MemoryInput->MemoryOutput pair, that means that the state is immediately populated with the init
+    // subgraph values when the init subgraph exists. In all the other cases the state is simply a read only object.
+    // We can optimize such a case removing the MemoryOutput node and transferring the state values update
+    // responsibility to a special type of the MemoryInput node - MemoryInputSingle
+    auto& graphNodes = graph.GetNodes();
+
+    auto isSuitableMemInput = [](const NodePtr& node) -> bool {
+        if (Type::MemoryInput != node->getType()) {
+            return false;
+        }
+        NodePtr MemoryOutput = nullptr;
+        auto&& childEdges = node->getChildEdgesAtPort(0);
+        for (auto&& item : childEdges) {
+            auto childNode = item->getChild();
+
+            if (Type::MemoryOutput == childNode->getType()) {
+                if (MemoryOutput && MemoryOutput != childNode) {
+                    //only one child MemoryOutput is expected
+                    return false;
+                }
+                MemoryOutput = childNode;
+            }
```

Comment:
Is it possible there is a case that ReadValue follows 2 Assginns, only 1 Assgin have same variable ID with current ReadValue?
If yes, I suggest only check if exist ReadValue follow Assgin with same Variable Id. What's your idea?

Reply:
Probably it does make sense, you're right. I will extend the condition.

Reply:
Done